networkx/algorithms/tests/test_similarity.py

```diff
@@ -982,3 +982,120 @@ def user_match(x, y):
             nx.graph_edit_distance(G3, G2, node_match=user_match, edge_match=user_match)
             == 1
         )
+
+    def test_panther_vector_similarity_basic(self):
+        """Basic test for panther_vector_similarity function."""
+        np.random.seed(42)
+
+        G = nx.Graph()
+        G.add_edge(0, 1)
+        G.add_edge(0, 2)
+        G.add_edge(0, 3)
+        G.add_edge(1, 2)
+        G.add_edge(2, 4)
+
+        sim = nx.panther_vector_similarity(G, 0, D=3, k=4, path_length=2)
+
+        assert len(sim) > 0
+        assert 0 not in sim  # Source node should not be included
+        assert all(node in [1, 2, 3, 4] for node in sim)  # Only valid nodes
+        assert all(0 <= score <= 1 for score in sim.values())  # Valid scores
+
+    def test_panther_vector_similarity_unweighted(self):
+        """Test panther_vector_similarity with unweighted graph."""
+        G = nx.Graph()
+        G.add_edge(0, 1)
+        G.add_edge(0, 2)
+        G.add_edge(0, 3)
+        G.add_edge(1, 2)
+        G.add_edge(2, 4)
+
+        sim = nx.panther_vector_similarity(G, 0, D=3, k=4, path_length=2)
+
+        assert len(sim) == 3
+
+        assert 0 not in sim
+
+        for node in [1, 2, 3]:
+            assert node in sim
+
+        for node, score in sim.items():
+            assert 0 <= score <= 1
```

Comment:
I like how this looks in the function above
```suggestion
        assert len(sim) == 3
        assert 0 not in sim
        assert all(node in sim for node in [1, 2, 3])
        assert all(0 <= score <= 1 for score in sim.values())
```